HPXMLtoOpenStudio/resources/EPvalidator.xml

```diff
@@ -1182,6 +1146,30 @@
     </sch:rule>
   </sch:pattern>
 
+  <sch:pattern name='[ApplianceLocation]'>
+    <sch:rule abstract='true' id='appliance_location'>
+	    <sch:assert role='ERROR' test='contains("_living space_ _basement - unconditioned_ _basement - conditioned_ _attic - unvented_ _attic - vented_ _garage_ _crawlspace - unvented_ _crawlspace - vented_ _other exterior_ _other housing unit_ _other heated space_ _other multifamily buffer space_ _other non-freezing space_", concat("_", text(), "_"))'>Expected "text()" for xpath: "living space", "basement - unconditioned", "basement - conditioned", "attic - unvented", "attic - vented", "garage", "crawlspace - unvented", "crawlspace - vented", "other exterior", "other housing unit", "other heated space", "other multifamily buffer space", or "other non-freezing space"</sch:assert>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:ClothesWasher/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:ClothesDryer/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:Dishwasher/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:Refrigerator/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:Freezer/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/h:CookingRange/h:Location'>
+      <sch:extends rule='appliance_location'/>
+    </sch:rule>
+  </sch:pattern>
+
```

Comment:
@shorowit what do you think this abstract rule approach to gather multiple rules and place them in one pattern? I guess this would make future maintenance a little easier.

Reply:
I like the idea of a single pattern. Thought the abstract rule definitely adds some complexity. A different approach would be to have a single rule w/ a wildcard, e.g.:

`context='/h:HPXML/h:Building/h:BuildingDetails/h:Appliances/*/h:Location'` (or `h:*`, whatever the right syntax is)

Reply:
You're right. I prefer the wildcard approach to the abstract rule approach. I'll make that change. Thanks!

Reply:
Right, that's the downside. I don't have a strong preference between A) using wildcards like you have to minimize the potential of inconsistent rules/enumerations between appliances, or B) having a separate rule for each appliance so that the context is more clear. I assume if we went with B, we could still have all the rules within a single pattern.

But I do prefer either of these approaches to using an abstract rule which just seems overly complicated.

Reply:
I agree. An abstract rule is overly complicated. I vote for Option B. I think providing clear warning messages is more important than simplifying the code. This should also be more clear for developers.